#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_use-sensible-browser.dpatch by Frank S. Thomas <frank@thomas-alfeld.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch is Debian specific. Don't use it for non Debian
## DP: packages. The program "sensible-browser" is part of the essential
## DP: package debianutils.

@DPATCH@
--- boinc-5.7.4.orig/clientgui/hyperlink.cpp	2006-10-27 12:26:55.000000000 +0200
+++ boinc-5.7.4/clientgui/hyperlink.cpp	2006-11-17 15:59:10.000000000 +0100
@@ -171,7 +171,8 @@
 }
 
 void wxHyperLink::ExecuteLink (const wxString &strLink) {
-    if (!wxLaunchDefaultBrowser(strLink)) {
+    wxString cmd = wxT("sensible-browser ") + strLink;
+    if (!::wxExecute(cmd, wxEXEC_ASYNC)) {
         wxString strDialogTitle = wxEmptyString;
         wxString strDialogMessage = wxEmptyString;
 
